Reuse types in _named_data_store and support tensor layouts

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #14667

Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

lucylq committed:
Reuse types in _named_data_store and support tensor layouts

Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]

lucylq committed:
Update on "Reuse types in _named_data_store and support tensor layouts"


Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]

lucylq committed:
Update on "Reuse types in _named_data_store and support tensor layouts"


Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]

larryliu0820 reviewed (changes requested):
Review automatically exported from Phabricator review in Meta.

lucylq committed:
Update on "Reuse types in _named_data_store and support tensor layouts"


Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]

lucylq committed:
Update on "Reuse types in _named_data_store and support tensor layouts"


Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]

lucylq committed:
Update on "Reuse types in _named_data_store and support tensor layouts"


Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]

larryliu0820 reviewed (changes requested):
Review automatically exported from Phabricator review in Meta.

lucylq committed:
Update on "Reuse types in _named_data_store and support tensor layouts"


Reuse `DataEntry` from data_serializer.py, in _named_data_store.py.

Motivation
- deserialize from flat tensor to named data store output
- support tensor layout in named data store

Differential Revision: [D83490345](https://our.internmc.facebook.com/intern/diff/D83490345/)

[ghstack-poisoned]
